### ISSUE TITLE
Not able to convert detectron2 model to TorchScript

### ISSUE THREAD

**Issue by Ma1hias** [facebookresearch/detectron2#3861]
Hello,

I trained a model ["Misc/cascade_mask_rcnn_X_152_32x8d_FPN_IN5k_gn_dconv.yaml"](https://github.com/facebookresearch/detectron2/blob/main/configs/Misc/cascade_mask_rcnn_X_152_32x8d_FPN_IN5k_gn_dconv.yaml) from the model zoo and want to convert it to TorchScript using the tracing method inside a Jupyter notebook. Following the [documentation](https://detectron2.readthedocs.io/en/latest/tutorials/deployment.html) I tried to export the model, but it did not work out. 

Exporting another model using the exact same code worked fine.

It seems that there are problems with the deform_conv layer but the tracing worked fine, just saving is not possible.  
Does this mean, that the export for this model is not supported yet? And if so, is there a way to export it?

Because of the [documentation](https://detectron2.readthedocs.io/en/latest/tutorials/deployment.html) it seems that all models can be exported using the tracing method.

What am I missing?
Thank you in advance.

## Instructions To Reproduce the 🐛 Bug:
1. Full runnable code:
```python
from detectron2 import model_zoo
from detectron2.utils.testing import get_sample_coco_image
from detectron2.modeling import GeneralizedRCNN
from detectron2.export import TracingAdapter, dump_torchscript_IR
import torch
import numpy as np

model = model_zoo.get("Misc/cascade_mask_rcnn_X_152_32x8d_FPN_IN5k_gn_dconv.yaml", trained=True)

test_image = get_sample_coco_image()
inputs = [{"image": test_image}] 

if isinstance(model, GeneralizedRCNN):
    print("is instance of GeneralizedRCNN")
    def inference(model, inputs):
        # use do_postprocess=False so it returns ROI mask
        inst = model.inference(inputs, do_postprocess=False)[0]
        return [{"instances": inst}]
else:
    inference = None
traceable_model = TracingAdapter(model, inputs, inference)
traceable_model.eval()
ts_model = torch.jit.trace(traceable_model, (test_image,))

with open("model.ts", "wb") as f:
    torch.jit.save(ts_model, f)
```
2. What exact command you run:
I run the above code inside a Jupyter notebook inside the posted environment. The error happens when trying to save the model to disk.
 
Using the [export_model.py](https://github.com/facebookresearch/detectron2/blob/main/tools/deploy/export_model.py) script throws the same error. 

3. __Full logs__ or other relevant observations:
```
RuntimeError: 
Could not export Python function call '_DeformConv'. Remove calls to Python functions before export. Did you forget to add @script or @script_method annotation? If this is a nn.ModuleList, add it to __constants__:
```

## Expected behavior:

Saving a TorchScript model to disk.

## Environment:
This is the CPU only environment, I tried it with the cuda enabled one too.
```
---------------------  --------------------------------------------------------------------------------------
sys.platform           linux
Python                 3.9.5 (default, Nov 18 2021, 16:00:48) [GCC 10.3.0]
numpy                  1.22.0
detectron2             0.6 @/home/Mathias/repos/detectron2_experiments/.venv/lib/python3.9/site-packages/detectron2
Compiler               GCC 7.3
CUDA compiler          not available
DETECTRON2_ENV_MODULE  <not set>
PyTorch                1.10.0+cpu @/home/Mathias/repos/detectron2_experiments/.venv/lib/python3.9/site-packages/torch
PyTorch debug build    False
GPU available          No: torch.cuda.is_available() == False
Pillow                 9.0.0
torchvision            0.11.1+cpu @/home/Mathias/repos/detectron2_experiments/.venv/lib/python3.9/site-packages/torchvision
fvcore                 0.1.5.post20211023
iopath                 0.1.9
cv2                    4.5.5
---------------------  --------------------------------------------------------------------------------------
PyTorch built with:
  - GCC 7.3
  - C++ Version: 201402
  - Intel(R) Math Kernel Library Version 2020.0.0 Product Build 20191122 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v2.2.3 (Git Hash 7336ca9f055cf1bfa13efb658fe15dc9b41f0740)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - LAPACK is enabled (usually provided by MKL)
  - NNPACK is enabled
  - CPU capability usage: AVX2
  - Build settings: BLAS_INFO=mkl, BUILD_TYPE=Release, CXX_COMPILER=/opt/rh/devtoolset-7/root/usr/bin/c++, CXX_FLAGS= -Wno-deprecated -fvisibility-inlines-hidden -DUSE_PTHREADPOOL -fopenmp -DNDEBUG -DUSE_KINETO -DLIBKINETO_NOCUPTI -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DSYMBOLICATE_MOBILE_DEBUG_HANDLE -DEDGE_PROFILER_USE_KINETO -O2 -fPIC -Wno-narrowing -Wall -Wextra -Werror=return-type -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-sign-compare -Wno-unused-parameter -Wno-unused-variable -Wno-unused-function -Wno-unused-result -Wno-unused-local-typedefs -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Wno-stringop-overflow, LAPACK_INFO=mkl, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, TORCH_VERSION=1.10.0, USE_CUDA=0, USE_CUDNN=OFF, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=OFF, USE_NNPACK=ON, USE_OPENMP=ON, 
```

**Comment by ppwwyyxx** [comment]
Models with deformable conv cannot be exported now. This is expected.

**Comment by awsankur** [comment]
Is there an update to the above? I am trying to trace a Detectron2 model with:

`n_mod_trace = torch.jit.trace(model, example_inputs=[inputs])`

But get the error: `:Dictionary inputs to traced functions must have consistent type. Found Tensor and int`

When I try to script the model with:
`scripted_model = torch.jit.script(model)`

I get the error:
`*** RuntimeError: 
Module 'ResNet' has no attribute 'stages' (This attribute exists on the Python module, but we failed to convert Python type: 'list' to a TorchScript type. Could not infer type of list element: Cannot infer concrete type of torch.nn.Module. Its type was inferred; try adding a type annotation for the attribute.):
  File "/home/ec2-user/anaconda3/envs/detectron/lib/python3.9/site-packages/detectron2/modeling/backbone/resnet.py", line 448
        if "stem" in self._out_features:
            outputs["stem"] = x
        for name, stage in zip(self.stage_names, self.stages):
                                                 ~~~~~~~~~~~ <--- HERE
            x = stage(x)
            if name in self._out_features:`
